contrib/initramfs/zfsunlock

```diff
@@ -0,0 +1,43 @@
+#!/bin/sh
+
+set -eu
+echo "Waiting for zfs init script to complete, press Ctrl-C to exit into shell ..."
+while [ ! -e /run/zfs_fs_name ]; do
+	if [ -e /run/zfs_unlock_complete ]; then
+		exit 0
+	fi
+	sleep 0.5
+done
+echo
+echo "Unlocking encrypted ZFS filesystems ..."
+echo "Enter the password or press Ctrl-C to exit into shell."
+echo
+zfs_fs_name=""
+fs_password_accepted=false
+mkfifo /run/zfs_unlock_complete_notify
+while true; do
+	while [ "$fs_password_accepted" = "true" ] && [ "$(cat /run/zfs_fs_name)" = "$zfs_fs_name" ] && [ ! -e /run/zfs_unlock_complete ]; do
+		sleep 0.5
+	done
+	if [ -e /run/zfs_unlock_complete ]; then
+		echo "Unlocking complete, resuming boot sequence."
+		echo "Please reconnect in a while."
+		echo "ok" > /run/zfs_unlock_complete_notify
```

Comment:
There is a racing condition once console input is killed, and several tests confirmed the dropbear disconnects before text about being disconnection is displayed to the user. This creates false assumption about the session crash. I have added pipeline wait/notify to prevent this.

Reply:
That makes sense. Is there a particular reason some of this uses the creation/existence of a file as the flag and this thing uses a FIFO? Can they all be done in the same way? If not, that's fine.

Reply:
wait/notify is sure better than checking flags, but the problem here is user hitting Ctrl-C during the password input at random intervals. After several use cases I've removed pipelines for syncronization during the password input, because I've discovered hitting Ctrl-C sometimes led to  pipeline stuck waiting for input. So, I chose to use wait/notify only when all interactions are done, i.e. on final stage.